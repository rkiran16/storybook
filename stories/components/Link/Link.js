import { html } from "lit-html";


export const Link = ({label = "Link action",href = "http://www.cardinalhealth.com",target = "_self", showIcon = false, inline = false, additionalClass = ""}) => {
    const linkWithIcon = () => html`
         <a href=${href} target=${target} class="d-inline-flex align-items-center text-decoration-none">
            <span class="text-decoration-underline">${label}</span>
            <span class="material-icons ms-1 mt-1 md-16">open_in_new</span>
         </a>`
    

    if (inline) {
        return html `<a href=${href} class=${`inline ${additionalClass}`} target=${target}>${label}</a>`;
    } else {
        return showIcon ? linkWithIcon() : html`<a href=${href} target=${target}>${label}</a>` ;
    }
}