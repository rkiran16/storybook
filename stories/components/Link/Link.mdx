import { Canvas, Meta, Story } from "@storybook/addon-docs";

# Link

## Usage

Links are used as navigational elements and can be used on their own or inline with text.

Use links when you want users to:

- Navigate to a different page within the application
- Navigate to an entirely different site
- Jump to an element on the same page
- Link to emails or phone numbers

We recommend links be three words or fewer. Because links take users to a new location,
it is important that their labels accurately reflect the content users will find at the link destination.
Use meaningful labels for links and avoid terms like “click here” or the web address itself.

<table class="table table-striped">
  <thead>
    <tr>
      <th scope="col">Variant</th>
      <th scope="col">Purpose</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Standalone</td>
      <td>
        Standalone links are used on their own directly following content. They
        should not be used within sentences or paragraphs. Standalone links
        replace the current page but the user stays within the same browser tab
      </td>
    </tr>
    <tr>
      <td>Standalone with icon</td>
      <td>
        Opens a new browser tab to either go to another page within the same
        site or go to an entirely new site. Strive to pull these out from body
        copy for easy scan-ability.
      </td>
    </tr>
    <tr>
      <td>Inline</td>
      <td>
        Inline links are used in sentences or paragraphs of text. Inline links
        should not be used on their own and should not be paired with icons.
      </td>
    </tr>
  </tbody>
</table>

## Link Formatting & Interactions

<br />
<br />

<div class="container">
  <div class="row">
    <div class="col">
      <h5>Link - Standalone</h5>
      <Story id="components-link--standalone" name="StandAlone" />
    </div>
    <div class="col">
      <h5>Link - with Icon</h5>
      <Story
        id="components-link--standalone-with-icon"
        name="StandAlone with Icon"
      />
    </div>
    <div class="col">
      <h5>Link - inline</h5>
      <Story id="components-link--inline" name="Inline" />
    </div>
  </div>
</div>

<br />
<br />

## Colored Links

<p>
  You can use the <code>.link-_</code> classes to colorize links. Unlike the{" "}
  <code>.text-_</code> classes, these classes have a <code>:hover</code> and{" "}
  <code>:focus</code> state.
</p>

<Canvas>
  <Story id="components-link--colored-links"></Story>
</Canvas>
