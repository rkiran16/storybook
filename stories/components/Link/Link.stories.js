import { Link } from "./Link";
import docs from "./Link.mdx";
import { html} from "lit-html";


export default {
   title: "Components/Link",
   parameters: {
      zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6298e3fb7bdcdea0c623a096",
      docs: {
         page: docs
      }
   },
   argTypes: {
      label: {
        name: 'label',
        type: { name: 'string', required: true },
        defaultValue: 'Link action',
        description: 'configure label for link',
        control: {
          type: 'text'
        }
      },
      href: {
         name: "href",
         type: { name: 'string', required: true },
         defaultValue: 'http://www.cardinalhealth.com',
         description: "Provide the href attribute for the <a> node",
         table: {
            type: { summary: 'string' }
          },
         control: {
            type: "text"
         }
      },
      target: {
         name: "target",
         control: { type: 'select' },
         description: "Define where to display the linked URL",
         options: ['_blank', '_self', '_parent', '_top'],
         defaultValue: "_self",
         table: {
            type: {  summary: "string & HTMLAttributeAnchorTarget"},
            defaultValue: { summary: '_self' },
         }
      },
      showIcon: {
         type: { name: 'boolean', required: false},
         description: "Show Icon",
         defaultValue: false,
         table: {
           type: { summary: 'boolean' },
           defaultValue: { summary: false },
         },
         control: {
           type: 'boolean'
         }
      }
   }
}

const Template = args => html`${Link(args)}`;

const coloredTemplate = args => {
   return html`
      <div class="d-flex">
            <a href="#" class="link-primary me-3">Primary link</a>
            <a href="#" class="link-secondary me-3">Secondary link</a>
            <a href="#" class="link-success me-3">Success link</a>
            <a href="#" class="link-danger me-3">Danger link</a>
            <a href="#" class="link-warning me-3">Warning link</a>
            <a href="#" class="link-info me-3">Info link</a>
            <a href="#" class="link-dark me-3">Dark link</a>
      </div>
   `
}

export const standalone = Template.bind({});

export const standaloneWithIcon = Template.bind({});
standaloneWithIcon.args = {
    showIcon: true
}

export const inline = Template.bind({});
inline.args = {
   inline: true
}

export const colored_links = coloredTemplate.bind({});
