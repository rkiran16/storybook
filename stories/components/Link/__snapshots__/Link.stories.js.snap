// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Link Colored Links 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
      <div class=\\"d-flex\\">
            <a href=\\"#\\" class=\\"link-primary me-3\\">Primary link</a>
            <a href=\\"#\\" class=\\"link-secondary me-3\\">Secondary link</a>
            <a href=\\"#\\" class=\\"link-success me-3\\">Success link</a>
            <a href=\\"#\\" class=\\"link-danger me-3\\">Danger link</a>
            <a href=\\"#\\" class=\\"link-warning me-3\\">Warning link</a>
            <a href=\\"#\\" class=\\"link-info me-3\\">Info link</a>
            <a href=\\"#\\" class=\\"link-dark me-3\\">Dark link</a>
      </div>
   ",
  ],
  "values": Array [],
}
`;

exports[`Storyshots Components/Link Inline 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "<a href=",
        " class=",
        " target=",
        ">",
        "</a>",
      ],
      "values": Array [
        "http://www.cardinalhealth.com",
        "inline ",
        "_self",
        "Link action",
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Link Standalone 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "<a href=",
        " target=",
        ">",
        "</a>",
      ],
      "values": Array [
        "http://www.cardinalhealth.com",
        "_self",
        "Link action",
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Link Standalone With Icon 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
         <a href=",
        " target=",
        " class=\\"d-inline-flex align-items-center text-decoration-none\\">
            <span class=\\"text-decoration-underline\\">",
        "</span>
            <span class=\\"material-icons ms-1 mt-1 md-16\\">open_in_new</span>
         </a>",
      ],
      "values": Array [
        "http://www.cardinalhealth.com",
        "_self",
        "Link action",
      ],
    },
  ],
}
`;
